mkosi/__init__.py

```diff
@@ -1619,20 +1619,23 @@ def run_ukify(
         else:
             cmd += ["--secureboot-certificate", context.config.secure_boot_certificate]
 
-    run(
+    result = run(
         cmd,
         stdin=(
             sys.stdin
             if context.config.secure_boot_key_source.type != KeySourceType.file
             else subprocess.DEVNULL
         ),
+        stdout=subprocess.PIPE,
```

Comment:
This should be optional and only done if we need the json output, so add a `json` argument to `run_ukify()` that translates to `"--json=short",` and only make stdout a pipe if that is passed.

You can then make the return type of the function `dict[str, Any]`, parse the json in `run_ukify()` itself if requested and return an empty dict if no json output was requested.

Reply:
How do I solve this one:

```
mkosi/__init__.py:1645:9: error: Returning Any from function declared to return
"dict[str, Any]"  [no-any-return]
            return json.loads(result.stdout.strip())
```

Reply:
Just use `cast()` to cast it to the right type.